libsignal-service/src/cipher.rs

```diff
@@ -184,27 +235,51 @@ struct Plaintext {
     data: Vec<u8>,
 }
 
+fn add_padding(version: u32, contents: &[u8]) -> Result<Vec<u8>, ServiceError> {
+    if version < 2 {
+        Err(ServiceError::InvalidFrameError {
+            reason: format!("Unknown version {}", version),
+        })
+    } else if version == 2 {
+        Ok(contents.to_vec())
+    } else {
```

Comment:
Is this how it's padded in Java? Seems weird that *>2 goes the same way, no?

Reply:
Hmm, I notice it's the same in `strip_padding`, which I translated from Java...

Reply:
Yes, I also translated this from Java :smile: